filebeat/crawler/prospector_log.go

```diff
@@ -251,14 +254,13 @@ func (p *ProspectorLog) checkExistingFile(h *harvester.Harvester, newFile *input
 		// We only need to keep it for the remainder of this iteration then we can assume it was deleted and forget about it
 		p.missingFiles[h.Path] = oldFile.FileInfo
 
-	} else if h.Stat.Finished() && oldFile.FileInfo.ModTime() != h.Stat.Fileinfo.ModTime() {
+	} else if h.State.Finished() {
```

Comment:
One more note here for the harvester overlapping: The idea is that the prospector will have for each file a prospector and will be able to track the state of the harvester over a channel (running / stopped). A new harvester on an existing file will only be started when the harvester is finished to prevent duplicates.